module_common: handle None value for templar

pilou- opened the pull request:
##### SUMMARY

Fix the following error introduced by b455901:

    $ ./hacking/test-module -m ./lib/ansible/modules/system/ping.py -I ansible_python_interpreter=/usr/bin/python
    Traceback (most recent call last):
      File "./hacking/test-module", line 268, in <module>
        main()
      File "./hacking/test-module", line 249, in main
        (modfile, modname, module_style) = boilerplate_module(options.module_path, options.module_args, interpreters, options.check, options.filename)
      File "./hacking/test-module", line 152, in boilerplate_module
        task_vars=task_vars
      File "ansible/lib/ansible/executor/module_common.py", line 910, in modify_module
        environment=environment)
      File "ansible/lib/ansible/executor/module_common.py", line 736, in _find_module_utils
        shebang, interpreter = _get_shebang(u'/usr/bin/python', task_vars, templar)
      File "ansible/lib/ansible/executor/module_common.py", line 452, in _get_shebang
        interpreter = templar.template(task_vars[interpreter_config].strip())
    AttributeError: 'NoneType' object has no attribute 'template'

Unit test provided.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 lib/ansible/executor/module_common.py

##### ANSIBLE VERSION
```
ansible 2.6.0 (devel 076ea07bb5) last updated 2018/02/23 19:40:33 (GMT +200)
```

##### ADDITIONAL INFORMATION

Remote pair programming fix, thanks to @mscherer for being the observer.

pilou- commented:
Unrelated error with `osx/10.11/2` instance:

> 2018-02-23 19:11:46 TASK [subversion : initial checkout] ***
> 2018-02-23 19:11:47 fatal: [testhost]: FAILED! => {"changed": false, "cmd": "/usr/bin/svn --non-interactive --trust-server-cert --no-auth-cache checkout -r HEAD https://github.com/jimi-c/test_role /var/root/ansible_testing/svn", "msg": "svn: E175002: Unable to connect to a repository at URL 'https://github.com/jimi-c/test_role'\nsvn: E175002: OPTIONS of 'https://github.com/jimi-c/test_role': SSL handshake failed: SSL error: tlsv1 alert protocol version (https://github.com)", "rc": 1, "stderr": "svn: E175002: Unable to connect to a repository at URL 'https://github.com/jimi-c/test_role'\nsvn: E175002: OPTIONS of 'https://github.com/jimi-c/test_role': SSL handshake failed: SSL error: tlsv1 alert protocol version (https://github.com)\n", "stderr_lines": ["svn: E175002: Unable to connect to a repository at URL 'https://github.com/jimi-c/test_role'", "svn: E175002: OPTIONS of 'https://github.com/jimi-c/test_role': SSL handshake failed: SSL error: tlsv1 alert protocol version (https://github.com)"], "stdout": "", "stdout_lines": []}

pilou- commented:
cc @abadger

abadger commented:
templar is not an optional part of the API anymore so I think we have to change test-module to create and pass in a templar.

pilou- committed:
module_common: set required parameter templar

Fix the following error (related to b455901):

  $ ./hacking/test-module -m ./lib/ansible/modules/system/ping.py -I ansible_python_interpreter=/usr/bin/python
  Traceback (most recent call last):
    File "./hacking/test-module", line 268, in <module>
      main()
    File "./hacking/test-module", line 249, in main
      (modfile, modname, module_style) = boilerplate_module(options.module_path, options.module_args, interpreters, options.check, options.filename)
    File "./hacking/test-module", line 152, in boilerplate_module
      task_vars=task_vars
    File "ansible/lib/ansible/executor/module_common.py", line 910, in modify_module
      environment=environment)
    File "ansible/lib/ansible/executor/module_common.py", line 736, in _find_module_utils
      shebang, interpreter = _get_shebang(u'/usr/bin/python', task_vars, templar)
    File "ansible/lib/ansible/executor/module_common.py", line 452, in _get_shebang
      interpreter = templar.template(task_vars[interpreter_config].strip())
  AttributeError: 'NoneType' object has no attribute 'template'

mscherer commented:
So, that mean we also need to remove the default value of None ?

pilou- commented:
This would imply to change order of the parameters.

abadger commented:
/me asking for other opinions among committers.

abadger commented:
Committers generlly seem to agree that modify_modules() is internal functionality so we are free to change the order of parameters and make it a positinal parameter.

pilou- committed:
module_common.modify_module: templar is required

mscherer commented:
shipit

pilou- commented:
ping @abadger

caphrim007 commented:
shipit

abadger commented:
rebuild_merge
